### PR TITLE
feat(primitives): drop node14

### LINKED ISSUE #537: Primitives: event-target-shim causes issues w/ edge-runtime running on top of node
## Bug Report

I lately ran into [an issue](https://github.com/connectrpc/connect-es/issues/749) which prevents using [`connect-es`](https://github.com/connectrpc/connect-es) in edge functions, including Next.js' middleware, due to [an issue with event-target-shim](https://github.com/mysticatea/event-target-shim/issues/61) which is used in `@edge-runtime/primitives`.

As this ponyfill is used for local dev environments and standalone builds of Next.js, the problem only becomes evident on these setups.

**Current behavior**

> Copied from referenced issues.

I try to use `connect-web` in the Next.js middleware. As Node APIs are not available there, using `connect-node` won't work.

I expect it to work, as the edge runtime seems to support all necessary features required by connect-web.

However, requests fail with the following error, leading me through `edge-runtime` into the packaged `event-target-shim`

```
- error Error [TypeError]: Cannot read properties of undefined (reading 'flags')
    at setRemoved (file:///app/node_modules/next/dist/compiled/edge-runtime/index.js:1:970334)
    at removeListenerAt (file:///app/node_modules/next/dist/compiled/edge-runtime/index.js:1:970334)
    at EventTarget.dispatchEvent (file:///app/node_modules/next/dist/compiled/edge-runtime/index.js:1:970334)
    at abortSignalAbort (file:///app/node_modules/next/dist/compiled/edge-runtime/index.js:1:970334)
    at AbortController.abort (file:///app/node_modules/next/dist/compiled/edge-runtime/index.js:1:970334)
    at done (webpack-internal:///(middleware)/./node_modules/@bufbuild/connect/dist/esm/protocol/run-call.js:103:24)
    at eval (webpack-internal:///(middleware)/./node_modules/@bufbuild/connect/dist/esm/protocol/run-call.js:33:9)
...
```

**Expected behavior/code**

Requests do not fail because of removing event listeners. See referenced issues.

**Possible solution**

Probably these issues should be fixed upstream, however it does not look like [event-target-shim](https://github.com/mysticatea/event-target-shim) is actively maintained.

As the event target API [was added to node in v15](https://nodejs.org/api/events.html#eventtarget-and-event-api), becoming stable in v15.4, it's available as long as you're not running a lower version, so probably it's possible to just conditionally use this shim, limiting the prone setups to node<15. Causes deviating implementation in `@edge-runtime/ponyfill` however.

I don't think it's an issue how `connect-es` uses the event target API. The issue is caused when removing an event listener in the listener callback, effectively manually creating a once-listener. This does not seem like some odd behaviour to me.

### PR DESCRIPTION
**Pull request by Kikobeats**
It drops the following polyfills:

- [x] removed `event-target-shim` (use native Event & EventTarget)
- [x] removed `@stardazed/streams-text-encoding` (use native webStreams instead)
- [x] removed `web-streams-polyfill` (use native webStreams instead)
- [x] removed `@peculiar/webcrypto` (use `require('node:crypto').webcrypto` instead)
- [x] removed `aggregate-error-ponyfill`

Closes #537

### CODE REVIEW

File: packages/vm/tests/instanceof.test.ts [hunk 1]
```diff
@@ -4,6 +4,30 @@
 
 import { EdgeVM } from '../src'
 
+it('DOMException', async () => {
```
Comment (nuta) [inline packages/vm/tests/instanceof.test.ts:7]: question: Is this test related to this upgrade?

File: packages/primitives/src/primitives/load.js [hunk 2]
```diff
@@ -206,10 +196,10 @@ export function load(scopedContext = {}) {
     scopedContext: {
       global: { ...scopedContext },
       ...scopedContext,
-      ...streamsImpl,
       ...urlImpl,
       ...abortControllerImpl,
       ...eventsImpl,
```
Comment (nuta) [inline packages/primitives/src/primitives/load.js:201]: nit: I noticed `eventsImpl` is still being used in this line despite it's removed in other diffs. is this intentional?

**Review by nuta — COMMENTED** [review]
Nice! Left nitpicks 🙏

File: packages/vm/tests/instanceof.test.ts (hunk 1 above)
Comment (Kikobeats) [reply packages/vm/tests/instanceof.test.ts:7]: It's a nice-to-have test. The Node.js `DOMException` implementation is not extending from `Error`. I wanted to verify `new DOMException() instanceof Error === true` before and after shipping this.

Comment (Schniz) [reply packages/vm/tests/instanceof.test.ts:7]: The Node.js DOMException does extend Error:

```sh-session
⛧ node -p 'new DOMException() instanceof Error'
true
```

how did you fix it? You DM'd me that it doesn't work <picture data-single-emoji=":lol:" title=":lol:">:lol:</picture>

Comment (Schniz) [reply packages/vm/tests/instanceof.test.ts:7]: btw from which version of Node.js DOMException exists? I just checked the following

```sh-session
⛧ fnm use 16
Using Node v16.18.1
⛧ node -p 'new DOMException() instanceof Error'
[eval]:1
new DOMException() instanceof Error
^

ReferenceError: DOMException is not defined
```

File: packages/primitives/src/primitives/load.js (hunk 2 above)
Comment (Kikobeats) [reply packages/primitives/src/primitives/load.js:201]: That's because `eventsImpl` is just holding `FetchEvent` and `PromiseRejectionEvent` constructor, so if you don't need once of them then you don't need to use them at `eventsImpl` since the rest of constructors (ReadableStream, ReadableStreamBYOBReader, etc) are available in the global context

File: packages/vm/tests/instanceof.test.ts (hunk 1 above)
Comment (Kikobeats) [reply packages/vm/tests/instanceof.test.ts:7]: The `DOMException` built-in in Node.js is not exactly extending from `Error`:
https://github.com/nodejs/node/blob/62b2cf30f2d1326dde9d4bc047f5611f17c4a20f/lib/internal/per_context/domexception.js#L53

That isn't a problem, unless you use a VM <picture data-single-emoji=":sadblob:" title=":sadblob:">:sadblob:</picture> 

so what I did is to just keep our polyfill (which is pretty minimal) and in that way the inheritance is correct.

**Review by ecklf — APPROVED** [review]
